src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than zero');
+    }
+
+    const sides = [a, b, c].sort((x, y) => y - x);
+
+    if (sides[0] >= sides[1] + sides[2]) {
+      throw new Error(
+        'The longest side of a triangle cannot be greater' +
+          'than or equal to the sum of the other two sides',
+      );
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
 
-export class Triangle implements Figure {}
+export class Circle implements Figure {
+  shape: string = 'circle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw Error('Radius must be greater than zero');
```

Comment:
[CODE STYLE] - When throwing an error, it's recommended to provide a more detailed description in the error message. For instance, 'Radius must be greater than zero' could be more informative, like 'Invalid circle radius: Radius must be greater than zero.'